src/datasources/cache/cache.first.data.source.spec.ts

```diff
@@ -35,11 +36,78 @@ describe('CacheFirstDataSource', () => {
+
+  it('should return the data returned by the underlying network interface and cache it', async () => {
+    const targetUrl = faker.internet.url({ appendSlash: false });
+    const cacheDir = new CacheDir(faker.word.sample(), faker.word.sample());
+    const notFoundExpireTimeSeconds = faker.number.int();
+    const data = JSON.parse(fakeJson());
+    mockNetworkService.get.mockImplementation((url) => {
+      switch (url) {
+        case targetUrl:
+          return Promise.resolve({ data });
+        default:
+          return Promise.reject(`No matching rule for url: ${url}`);
+      }
+    });
+
+    const actual = await cacheFirstDataSource.get({
+      cacheDir,
+      url: targetUrl,
+      notFoundExpireTimeSeconds,
+    });
+
+    expect(actual).toEqual(data);
+    expect(mockNetworkService.get).toHaveBeenCalledTimes(1);
+    expect(fakeCacheService.keyCount()).toBe(1); // only data is cached (as no invalidation happened yet a this point in time)
+    expect(await fakeCacheService.get(cacheDir)).toEqual(JSON.stringify(data));
+  });
+
+  it('should return the network data and it should cache it if the last invalidation happened before the request was initiated', async () => {
     const targetUrl = faker.internet.url({ appendSlash: false });
     const cacheDir = new CacheDir(faker.word.sample(), faker.word.sample());
     const notFoundExpireTimeSeconds = faker.number.int();
     const data = JSON.parse(fakeJson());
+    const invalidationTimeMs = jest.now(); // invalidation happens at this point in time
+    fakeCacheService.set(
+      new CacheDir(`invalidationTimeMs:${cacheDir.key}`, ''),
+      invalidationTimeMs.toString(),
+    );
+    mockNetworkService.get.mockImplementation((url) => {
+      switch (url) {
+        case targetUrl:
+          return Promise.resolve({ data });
+        default:
+          return Promise.reject(`No matching rule for url: ${url}`);
+      }
+    });
+
+    jest.advanceTimersByTime(1); // the request is sent 1 ms after invalidation happened
+    const actual = await cacheFirstDataSource.get({
+      cacheDir,
+      url: targetUrl,
+      notFoundExpireTimeSeconds,
+    });
+
+    expect(actual).toEqual(data);
+    expect(mockNetworkService.get).toHaveBeenCalledTimes(1);
+    expect(fakeCacheService.keyCount()).toBe(2); // both data and invalidation timestamp are cached
+    expect(await fakeCacheService.get(cacheDir)).toEqual(JSON.stringify(data)); // item is cached
+  });
+
+  it("should return the network data but it shouldn't not cache it if the last invalidation happened after the request was initiated", async () => {
+    const targetUrl = faker.internet.url({ appendSlash: false });
+    const cacheDir = new CacheDir(faker.word.sample(), faker.word.sample());
+    const notFoundExpireTimeSeconds = faker.number.int();
+    const data = JSON.parse(fakeJson());
+    const invalidationTimeMs = jest.now() + 1; // invalidation happens 1 ms after this point in time
+    fakeCacheService.set(
```

Comment:
We are missing an `await` here (similarly in other tests)

Reply:
`FakeCacheService` functions are synchronous (unlike async implementations like `RedisCacheService`).

`FakeCacheService` functions are called with no `await` in this file as you said, `health.controller.spec.ts` or `flush.controller.spec.ts`. I think it makes no difference, but we could agree on a common approach and unify all the calls in a separate PR, wdyt?

Reply:
You are right, I'm going to add the `await` statement here, and  will create a separate PR to fix it in other classes 👍

Reply:
Done in [e4d030f](https://github.com/safe-global/safe-client-gateway/pull/970/commits/e4d030ffdc0a86bf23d2170be3f200856769c27e)